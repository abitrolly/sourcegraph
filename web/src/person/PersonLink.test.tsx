import React from 'react'
import { MemoryRouter } from 'react-router'
import { PersonLink } from './PersonLink'
import { mount } from 'enzyme'

describe('PersonLink', () => {
    test('no user account', () =>
        expect(
            mount(
                <PersonLink
                    person={{ displayName: 'alice', email: 'alice@example.com', user: null }}
                    className="a"
                    userClassName="b"
                />
            )
        ).toMatchSnapshot())

    test('with user account', () =>
        expect(
            mount(
                <MemoryRouter>
                    <PersonLink
                        person={{
                            displayName: 'Alice',
                            email: 'alice@example.com',
                            user: { username: 'alice', displayName: 'Alice Smith', url: 'u' },
                        }}
                        className="a"
                        userClassName="b"
                    />
                </MemoryRouter>
            )
        ).toMatchSnapshot())
})
