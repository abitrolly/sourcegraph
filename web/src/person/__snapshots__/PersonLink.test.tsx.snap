// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonLink no user account 1`] = `
<PersonLink
  className="a"
  person={
    Object {
      "displayName": "alice",
      "email": "alice@example.com",
      "user": null,
    }
  }
  userClassName="b"
>
  <LinkOrSpan
    className="a "
    data-tooltip="alice@example.com"
  >
    <span
      className="a "
      data-tooltip="alice@example.com"
    >
      alice
    </span>
  </LinkOrSpan>
</PersonLink>
`;

exports[`PersonLink with user account 1`] = `
<MemoryRouter>
  <Router
    history="[History]"
  >
    <PersonLink
      className="a"
      person={
        Object {
          "displayName": "Alice",
          "email": "alice@example.com",
          "user": Object {
            "displayName": "Alice Smith",
            "url": "u",
            "username": "alice",
          },
        }
      }
      userClassName="b"
    >
      <LinkOrSpan
        className="a b"
        data-tooltip="Alice Smith <alice@example.com>"
        to="u"
      >
        <AnchorLink
          className="a b"
          data-tooltip="Alice Smith <alice@example.com>"
          to="u"
        >
          <a
            className="a b"
            data-tooltip="Alice Smith <alice@example.com>"
            href="u"
          >
            alice
          </a>
        </AnchorLink>
      </LinkOrSpan>
    </PersonLink>
  </Router>
</MemoryRouter>
`;
